lib/Sema/CSSimplify.cpp

```diff
@@ -4168,8 +4177,18 @@ repairViaOptionalUnwrap(ConstraintSystem &cs, Type fromType, Type toType,
   // If this is a conversion to a non-optional contextual type e.g.
   // `let _: Bool = try? foo()` and `foo()` produces `Int`
   // we should diagnose it as type mismatch instead of missing unwrap.
+  //
+  // We should also handle argument conversion mismatch e.g.
```

Comment:
It would be better to explain "why" we need it instead of just stating that we do.